includes/class-wc-gateway-stripe.php

```diff
@@ -1089,4 +1089,33 @@ public function get_checkout_payment_url( $pay_url, $order ) {
 		}
 		return $pay_url;
 	}
+
+	/**
+	 * Checks whether new keys are being entered when saving options.
+	 */
+	public function process_admin_options() {
+		// Load all old values before the new settings get saved.
+		$old_publishable_key      = $this->get_option( 'publishable_key' );
+		$old_secret_key           = $this->get_option( 'secret_key' );
+		$old_test_publishable_key = $this->get_option( 'test_publishable_key' );
+		$old_test_secret_key      = $this->get_option( 'test_secret_key' );
+
+		parent::process_admin_options();
+
+		// Load all old values after the new settings have been saved.
+		$new_publishable_key      = $this->get_option( 'publishable_key' );
+		$new_secret_key           = $this->get_option( 'secret_key' );
+		$new_test_publishable_key = $this->get_option( 'test_publishable_key' );
+		$new_test_secret_key      = $this->get_option( 'test_secret_key' );
+
+		// Look for updates.
+		if (
+			( ! empty( $old_publishable_key ) && $old_publishable_key !== $new_publishable_key )
+			|| ( ! empty( $old_secret_key ) && $old_secret_key !== $new_secret_key )
+			|| ( ! empty( $old_test_publishable_key ) && $old_test_publishable_key !== $new_test_publishable_key )
+			|| ( ! empty( $old_test_secret_key ) && $old_test_secret_key !== $new_test_secret_key )
+		) {
+			update_option( 'wc_stripe_show_changed_keys_notice', 'yes' );
+		}
```

Comment:
I'm wondering if the `! empty( $key )` condition is required here? I don't see how it's relevant.
Is it to prevent the notice from showing before any keys are added? I.e. when both the old and new keys are still just an empty string?

One _super_ minor thing; I think this `if` statement would be easier to understand if we added a function like `is_different( $old_key, $new_key )` just for legibility. It's difficult to parse `if` statements like this one 🙂.

The function would only need to contain the same check:
```php
return ( ! empty( $old_key ) && $old_key !== $new_key );
```

We could even define it inline as a [static anonymous function](https://www.php.net/manual/en/functions.anonymous.php) if we feel like the `is_different` has no purpose outside this particular function.

Reply:
>I'm wondering if the `! empty( $key )` condition is required here? I don't see how it's relevant.
Is it to prevent the notice from showing before any keys are added? I.e. when both the old and new keys are still just an empty string?

_Duh_. Maybe I'll read the PR description next time 🙃.

My minor nitpick is still there though 🙂.

Reply:
> Duh. Maybe I'll read the PR description next time 🙃.

:p

> My minor nitpick is still there though 🙂.

Agreed and fixed in https://github.com/woocommerce/woocommerce-gateway-stripe/pull/1131/commits/2b83a184df4a31dec32928fca09546c6ad6ff4f1